ext/standard/php_uuid.h

```diff
@@ -0,0 +1,163 @@
+/*
+  +----------------------------------------------------------------------+
+  | PHP Version 7                                                        |
+  +----------------------------------------------------------------------+
+  | Copyright (c) 1997-2017 The PHP Group                                |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | http://www.php.net/license/3_01.txt                                  |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Richard Fussenegger <php@fleshgrinder.com>                   |
+  +----------------------------------------------------------------------+
+*/
+
+/* $Id$ */
+
+#ifndef PHP_UUID_H
+#define PHP_UUID_H
+
+#include "zend_types.h"
+#include "php.h"
+
+PHPAPI extern zend_class_entry *php_ce_UUID;
+PHPAPI extern zend_class_entry *php_ce_UUIDParseException;
+
+#define PHP_UUID_LEN 16
+typedef struct php_uuid {
+	uint8_t bytes[PHP_UUID_LEN];
+} php_uuid;
+
+#define PHP_UUID_HEX_LEN 33
+typedef struct php_uuid_hex {
+	char str[PHP_UUID_HEX_LEN];
+} php_uuid_hex;
+
+#define PHP_UUID_STRING_LEN 37
+typedef struct php_uuid_string {
+	char str[PHP_UUID_STRING_LEN];
+} php_uuid_string;
+
+typedef enum php_uuid_variant {
+	PHP_UUID_VARIANT_NCS             = 0,
+	PHP_UUID_VARIANT_RFC4122         = 1,
+	PHP_UUID_VARIANT_MICROSOFT       = 2,
+	PHP_UUID_VARIANT_FUTURE_RESERVED = 3,
+} php_uuid_variant;
+
+static const uint8_t PHP_UUID_VERSION_1_TIME_BASED      = 1;
+static const uint8_t PHP_UUID_VERSION_2_DCE_SECURITY    = 2;
+static const uint8_t PHP_UUID_VERSION_3_NAME_BASED_MD5  = 3;
+static const uint8_t PHP_UUID_VERSION_4_RANDOM          = 4;
+static const uint8_t PHP_UUID_VERSION_5_NAME_BASED_SHA1 = 5;
+
+BEGIN_EXTERN_C()
+
+PHPAPI void php_uuid_create_v3(php_uuid *uuid, const php_uuid *namespace, const char *name, const size_t name_len);
+
+PHPAPI int php_uuid_create_v4(php_uuid *uuid, const zend_bool throw);
+
+static zend_always_inline int php_uuid_create_v4_silent(php_uuid *uuid)
```

Comment:
Do we really need the non-throwing variation of this? random_bytes has it because it's also used in some places which are historically non-throwing, but this is a new API.

Reply:
This information is not available to other developers, because there is not a single comment nowhere. I thought that this is the way it has to be done, because it is done the same way in the random module. Of course we do not need it.